mods/gen4/scripts.js

```diff
@@ -233,21 +233,32 @@ exports.BattleScripts = {
 				case 'explosion': case 'selfdestruct':
 					if (counter.stab < 2 || counter.setupType || !!counter['recovery'] || hasMove['rest'] || hasMove['substitute']) rejected = true;
 					break;
-				case 'foresight': case 'protect': case 'roar':
+				case 'foresight': case 'roar':
 					if (counter.setupType && !hasAbility['Speed Boost']) rejected = true;
 					break;
+				case 'protect':
+					if (!(hasAbility['Speed Boost'] || hasAbility['Guts'] || hasAbility['Quick Feet'] || hasMove['Wish'] || hasMove['Toxic'])) rejected = true;
+					break;
+				case 'wish':
+					if (!(hasMove['Protect'] || hasMove['U-turn'] || hasMove['Baton Pass'])) rejected = true;
+					if (hasMove['rest'] || !!counter['speedsetup']) rejected = true;
+					break;
 				case 'rapidspin':
 					if (teamDetails.rapidSpin) rejected = true;
+					if (counter.setupType && counter.Physical + counter.Special < 2) rejected = true;
 					break;
 				case 'stealthrock':
 					if (counter.setupType || !!counter['speedsetup'] || hasMove['rest'] || teamDetails.stealthRock) rejected = true;
 					break;
+				case 'reflect': case 'lightscreen':
+					if (counter.setupType) rejected = true;
+					break;
 				case 'switcheroo': case 'trick':
 					if (counter.Physical + counter.Special < 3 || counter.setupType) rejected = true;
 					if (hasMove['lightscreen'] || hasMove['reflect'] || hasMove['suckerpunch'] || hasMove['trickroom']) rejected = true;
 					break;
 				case 'toxic': case 'toxicspikes':
-					if (counter.setupType || teamDetails.toxicSpikes) rejected = true;
+					if (isSetup || counter.setupType || teamDetails.toxicSpikes) rejected = true;
```

Comment:
Hmm, I'll leave it alone for now then. I don't really want to take either Toxic or the setup moves out of those Pokemons' movepools.